lib/wrapeth/provider.ex

```diff
@@ -170,16 +191,33 @@ defmodule Wrapeth.Provider do
 
       @impl true
       def eth_gas_price(_args \\ []) do
-        call_client(:eth_gas_price)
+        call_client(:eth_gas_price, [])
       end
 
       @impl true
       def eth_get_transaction_receipt(hash, _args \\ []) do
         call_client(:eth_get_transaction_receipt, [hash])
       end
 
-      defp call_client(method_name, args \\ [], _arg \\ []) do
-        apply(@client_type, method_name, args ++ [[{:url, @node_url}]])
+      @impl true
+      def eth_subscribe(envent_type \\ "newHeads") do
+        call_client(:eth_subscribe, [envent_type])
+      end
+
+      @impl true
+      def eth_unsubscribe(sub_id) do
+        call_client(:eth_unsubscribe, [sub_id])
+      end
+
+      defp call_client(method_name, args \\ [], _args \\ []) do
+        case @client_type do
+          WebSocket -> request(@method_name_map[method_name], args)
```

Comment:
It's because in Ethereumex, the method name is different from the client.

Reply:
Oh, that's because we're using snake case instead of camel case, right?
Do you prefer to have a map on top of the module, or do you prefer to have a function that transforms snake case into camel case?